### PR TITLE
fix: no sagemaker autostop lag

### PR DESCRIPTION
**Pull request by maghirardelli**
Issue #, if available: GALI-965

Description of changes: Reduces lag of Sagemaker instances showing up as available when they have autostopped or are autostopping to a few seconds from a few minutes (or infinity in some cases)

Checklist:

- [x] Have you successfully deployed to an AWS account with your changes?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully tested with your changes locally?

AS review ticket id:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/__tests__/environment-sc-service.test.js [hunk 1]
```diff
@@ -1422,4 +1422,112 @@ describe('EnvironmentSCService', () => {
...
+      // OPERATE
+      const sagemakerUpdated = await service.pollAndSyncSageMakerStatus(
+        roleArn,
+        externalId,
+        sagemakerInstances,
+        requestContext,
+      );
+
+      // CHECK
+      await expect(sagemakerUpdated).not.toEqual(empty);
```
Comment (nguyen102) [inline addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/__tests__/environment-sc-service.test.js]: Nitpick: What's the actual value that is returned here? Can we just check it returns that value?

File: addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/__tests__/environment-sc-service.test.js [hunk 2]
```diff
@@ -1422,4 +1422,112 @@ describe('EnvironmentSCService', () => {
...
+
+    it('should finish updating before returning', async () => {
+      // BUILD
+      const sagemakerInstances = { 'notebook-instance-name': {} };
+      service.pollSageMakerRealtimeStatus = jest.fn(() => {
+        return { 'notebook-instance-name': 'InService' };
+      });
+      service.updateStatus = jest.fn(async () => {
+        // sleep
+        await new Promise(r => setTimeout(r, 4000));
```
Comment (nguyen102) [inline addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/__tests__/environment-sc-service.test.js]: Can we make this `1000` and the original code without the bug fix will still fail this unit test? If yes, let's shorten the time frame so the test runs faster.

File: addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/__tests__/environment-sc-service.test.js (hunk 1 above)
Comment (maghirardelli) [reply addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/__tests__/environment-sc-service.test.js]: Oh yeah, I think we can do that too. I'll change

File: addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/__tests__/environment-sc-service.test.js (hunk 2 above)
Comment (maghirardelli) [reply addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/__tests__/environment-sc-service.test.js]: let me see. If so, I will make it shorter.

**Review by nguyen102 — APPROVED** [review]

**Pull request merged**